packages/validator/package.json

```diff
@@ -18,9 +18,11 @@
     "test": "jest",
     "validate": "ts-node src/commands/validateCmd.ts",
     "convert": "ts-node src/commands/convertCmd.ts",
-    "compile": "tsc -p tsconfig.build.json"
+    "compile": "tsc -p tsconfig.build.json",
+    "docs": "ts-node docs/genDocs.ts"
   },
   "devDependencies": {
-    "ts-node": "^9.0.0"
+    "ts-node": "^9.0.0",
+    "markdown-snippet-injector": "^0.2.4"
```

Comment:
I'm also very wary of adding a library that has gone ~3 years without a commit and averages 39 downloads/week...

Reply:
I spent quite a bit time trying out bigger and more popular libraries, however for various reasons none of them worked for this exact problem. This one might be not developed anymore however it is used purely for one simple task, which will be ran very rarely. Because of this I don't think using something not maintained is a big issue, other option would be writing something custom, but thats maybe a bit unnecessary if library doing this already exists...
Im also not sure if validator docs will be included in api3 docs, I didnt look how api3 docs are written but it might take quite a time to figure out an automatic way of keeping tests and docs examples in sync, so for now they should stay here and after more pressing issues are resolved I can take a look at api3 docs.

Reply:
> I spent quite a bit time trying out bigger and more popular libraries, however for various reasons none of them worked for this exact problem. This one might be not developed anymore however it is used purely for one simple task, which will be ran very rarely. Because of this I don't think using something not maintained is a big issue, other option would be writing something custom, but thats maybe a bit unnecessary if library doing this already exists...

Did you try https://github.com/DavidWells/markdown-magic? This one offers a bit more, and avoids the need to have the JSON objects in JS files and also avoids disabling the ESlint rule. I strongly disagree with writing custom (even the current injector does the job).

> Im also not sure if validator docs will be included in api3 docs, I didnt look how api3 docs are written but it might take quite a time to figure out an automatic way of keeping tests and docs examples in sync, so for now they should stay here and after more pressing issues are resolved I can take a look at api3 docs.

Yeah, I think my second suggestion makes the most sense and api3docs can link to the READMEs in the validator package. The disadvantage is that part of the docs will be outside of api3docs, but I think it's not a huge deal.